packages/headless/src/controllers/commerce/facets/core/headless-core-commerce-facet.ts

```diff
@@ -27,66 +31,93 @@ import {buildController} from '../../../controller/headless-controller';
 import {
   CoreFacet as HeadlessCoreFacet,
   CoreFacetState,
-  FacetValue,
+  FacetValue as RegularFacetValue,
 } from '../../../core/facets/facet/headless-core-facet';
+import {NumericFacetValue} from '../../../core/facets/range-facet/numeric-facet/headless-core-numeric-facet';
+
+export type {FacetValueState, RegularFacetValue, NumericFacetValue};
 
-export type {FacetValue, FacetValueState};
+export type AnyFacetValue =
+  | RegularFacetValue
+  | NumericFacetValue
+  | DateFacetValue
+  | CategoryFacetValue;
+
+interface AnyToggleFacetValueActionCreatorPayload {
+  selection: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  facetId: string;
+}
 
 /**
  * @internal
- * This prop is used internally by the `Facet` controller.
+ *
+ * The configurable `CoreCommerceFacet` properties used internally.
  */
-export interface FacetProps {
-  options: FacetOptions;
+export interface CoreCommerceFacetProps {
+  options: CoreCommerceFacetOptions;
 }
 
-export interface FacetOptions {
-  /**
-   * A unique identifier for the controller.
-   * */
+export interface CoreCommerceFacetOptions {
   facetId: string;
+  toggleSelectActionCreator?: ActionCreatorWithPreparedPayload<
```

Comment:
This is the big change here: the core facet controller accepts action creators.

I made them optional because I'm not sure how to make them pass schema validation... but since this will be 100% handled internally, I think we probably don't even need to validate (or publicly expose) the props of this controller at all.

Reply:
Is that for analytics mainly? or something else?